$NetBSD: patch-config_pre.in,v 1.3 2018/06/15 20:46:01 tez Exp $

Add --enable-pkgsrc-libtool option
Use $(ROOT_USER) in place of root
Make sure LDFLAGS are not ignored

--- config/pre.in.orig	2018-05-03 14:34:47.000000000 +0000
+++ config/pre.in
@@ -171,6 +171,7 @@ PTHREAD_LIBS = @PTHREAD_LIBS@
 THREAD_LINKOPTS = $(PTHREAD_CFLAGS) $(PTHREAD_LIBS)
 CPPFLAGS = @CPPFLAGS@
 DEFS = @DEFS@
+LIBTOOL = @LIBTOOL@
 CC = @CC@
 CXX = @CXX@
 LD = $(PURE) @LD@
@@ -180,11 +181,12 @@ LIBS = @LIBS@
 
 INSTALL=@INSTALL@
 INSTALL_STRIP=
-INSTALL_PROGRAM=@INSTALL_PROGRAM@ $(INSTALL_STRIP)
-INSTALL_SCRIPT=@INSTALL_PROGRAM@
+INSTALL_LIBTOOL=@INSTALL_LIBTOOL@
+INSTALL_PROGRAM=$(INSTALL_LIBTOOL) @INSTALL_PROGRAM@ $(INSTALL_STRIP)
+INSTALL_SCRIPT=@INSTALL_SCRIPT@
 INSTALL_DATA=@INSTALL_DATA@
 INSTALL_SHLIB=@INSTALL_SHLIB@
-INSTALL_SETUID=$(INSTALL) $(INSTALL_STRIP) -m 4755 -o root
+INSTALL_SETUID=$(INSTALL_LIBTOOL) $(INSTALL) $(INSTALL_STRIP) -m 4555 -o $(ROOT_USER)
 ## This is needed because autoconf will sometimes define @exec_prefix@ to be
 ## ${prefix}.
 prefix=@prefix@
@@ -315,6 +317,10 @@ PROG_RPATH_FLAGS=@PROG_RPATH_FLAGS@
 # depending on whether we're building with shared libraries.
 DEPLIBEXT=@DEPLIBEXT@
 
+# this gets set by configure to either $(STLIBEXT) or $(LALIBEXT),
+# depending on whether we're building libtool archives.
+DEPSTLIBEXT=@DEPSTLIBEXT@
+
 KDB5_PLUGIN_DEPLIBS = @KDB5_PLUGIN_DEPLIBS@
 KDB5_PLUGIN_LIBS = @KDB5_PLUGIN_LIBS@
 
@@ -332,12 +338,12 @@ COM_ERR_DEPLIB-k5 = $(TOPLIBD)/libcom_er
 SUPPORT_LIBNAME=krb5support
 SUPPORT_DEPLIB	= $(TOPLIBD)/lib$(SUPPORT_LIBNAME)$(DEPLIBEXT)
 
-# These are forced to use ".a" as an extension because they're never
+# These are forced to use "@DEPSTLIBEXT" as an extension because they're never
 # built shared.
 SS_DEPLIB	= $(SS_DEPLIB-@SS_VERSION@)
-SS_DEPLIB-k5	= $(TOPLIBD)/libss.a
+SS_DEPLIB-k5	= $(TOPLIBD)/libss$(DEPSTLIBEXT)
 SS_DEPLIB-sys	=
-APPUTILS_DEPLIB	= $(TOPLIBD)/libapputils.a
+APPUTILS_DEPLIB	= $(TOPLIBD)/libapputils$(DEPSTLIBEXT)
 
 KRB5_BASE_DEPLIBS	= $(KRB5_DEPLIB) $(CRYPTO_DEPLIB) $(COM_ERR_DEPLIB) $(SUPPORT_DEPLIB)
 KDB5_DEPLIBS		= $(KDB5_DEPLIB) $(KDB5_PLUGIN_DEPLIBS)
@@ -376,7 +382,7 @@ RL_LIBS		= @RL_LIBS@
 
 SS_LIB		= $(SS_LIB-@SS_VERSION@)
 SS_LIB-sys	= @SS_LIB@
-SS_LIB-k5	= $(TOPLIBD)/libss.a $(RL_LIBS)
+SS_LIB-k5	= $(TOPLIBD)/libss$(DEPSTLIBEXT) $(RL_LIBS)
 KDB5_LIB	= -lkdb5 $(KDB5_PLUGIN_LIBS)
 
 VERTO_DEPLIB	= $(VERTO_DEPLIB-@VERTO_VERSION@)
@@ -556,6 +562,9 @@ MAKE_DYNOBJ_COMMAND=@MAKE_DYNOBJ_COMMAND
 DYNOBJ_EXPDEPS=@DYNOBJ_EXPDEPS@
 DYNOBJ_EXPFLAGS=@DYNOBJ_EXPFLAGS@
 
+# usually .la
+LALIBEXT=@LALIBEXT@
+
 # For some platforms, a flag which causes shared library creation to
 # check for undefined symbols.  Suppressed when using --enable-asan.
 UNDEF_CHECK=@UNDEF_CHECK@
@@ -587,6 +596,7 @@ SHLIB_RPATH_FLAGS=@SHLIB_RPATH_FLAGS@
 # flags for explicit libraries depending on this one,
 # e.g. "$(SHLIB_RPATH_FLAGS) $(SHLIB_SHLIB_DIRFLAGS) $(SHLIB_EXPLIBS)"
 SHLIB_EXPFLAGS=@SHLIB_EXPFLAGS@
+LALIB_EXPFLAGS=@LALIB_EXPFLAGS@ $(LDFLAGS)
 
 ## Parameters to be set by configure for use in libobj.in:
 
@@ -598,6 +608,11 @@ OBJLISTS=@OBJLISTS@
 # the suffix substitution will break on some platforms!
 SHLIBOBJS=$(STLIBOBJS:.o=@SHOBJEXT@)
 PFLIBOBJS=$(STLIBOBJS:.o=@PFOBJEXT@)
+LALIBOBJS=$(STLIBOBJS:.o=@LAOBJEXT@)
+
+# Args to tack on the tail of LIBTOOL to generate versioned, installable
+# libtool archives.
+LIBTOOL_TAIL=@LIBTOOL_TAIL@
 
 #
 # rules to make various types of object files
